nengo/utils/testing.py

```diff
@@ -179,7 +179,7 @@ class WarningCatcher(object):
     def __enter__(self):
         self.catcher = warnings.catch_warnings(record=True)
         self.record = self.catcher.__enter__()
-        warnings.simplefilter('always', append=True)
+        warnings.simplefilter('always')
```

Comment:
We don't require users to install pytest, so upping the required version only affect developers who should have no problems upgrading. So I'm cool upping it, but yes, in another PR :)